systemlinkmigrate.py

```diff
@@ -1,17 +1,29 @@
 # Generic migration utility for migrating various data and settings between SystemLink servers.
 # Not all services will be supported. Addtional services will be supported over time.
 
-import sys
+import sys, os
 from slmigrate import mongohandler, filehandler, arghandler, servicemgrhandler, constants
 
+
 # Main
-if __name__ == "__main__":
-    arguments = arghandler.parse_arguments().parse_args()
+def main():
+    argparser = arghandler.parse_arguments()
+    arguments = argparser.parse_args()
     arghandler.determine_migration_dir(arguments)
+    services_to_migrate = arghandler.determine_migrate_action(arguments)
+    if arguments.action == constants.restore_arg and not filehandler.migration_dir_exists(constants.migration_dir):
+        argparser.error(constants.migration_dir + " does not exist")
+    for service_to_migrate in services_to_migrate:
```

Comment:
I think this code and the code on starting on line 27 could be more [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). We are doing the same for loop twice. You might also encapsulate the if statements and error handing in a function to keep this main function cleaner.

Reply:
I'm thinking about how to do this.  I think it takes some extra refactoring to do this cleanly and still have the desired behavior of not stopping all SL services and Mongo if we have bad input.

Reply:
I cleaned up main to make it a little bit more clear what is happening.  I still would ideally like to combine the two loops, but I don't think there's a clean way to do that at present.  I'm open to suggestions though.